apps/prairielearn/src/lib/groups.js

```diff
@@ -265,40 +250,28 @@ module.exports.joinGroup = function (joinCode, assessmentId, userId, authnUserId
  * @param {string} assessmentId
  * @param {string} userId
  * @param {string} authnUserId
- * @param {Function} callback
  */
-module.exports.createGroup = function (groupName, assessmentId, userId, authnUserId, callback) {
-  let params = {
-    assessment_id: assessmentId,
-    user_id: userId,
-    authn_user_id: authnUserId,
-    group_name: groupName,
-  };
+module.exports.createGroup = async function (groupName, assessmentId, userId, authnUserId) {
   //alphanumeric characters only
-  let invalidGroupName = true;
-  const letters = /^[0-9a-zA-Z]+$/;
-  if (groupName.length <= 30 && groupName.match(letters)) {
-    invalidGroupName = false;
-    sqldb.query(sql.create_group, params, function (err, _result) {
-      if (!err) {
-        callback(null, true);
-      } else {
-        sqldb.query(sql.get_group_config, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          const groupConfig = result.rows[0];
-          const uniqueGroupName = true;
-          callback(null, false, uniqueGroupName, null, groupConfig);
-        });
-      }
-    });
+  if (groupName.length > 30 || !groupName.match(/^[0-9a-zA-Z]+$/)) {
+    flash(
+      'error',
+      'The group name is invalid. Alpha and numeric characters only. ([0-9] [a-z] [A-Z]), with maximum length of 30 characters.',
```

Comment:
With full acknowledgement that you're just copy/pasting existing text: showing regex character classes here is kinda gross. Can we improve this?

Consider splitting this into two separate error cases:

- Too long ("The group name must be less than 30 characters.")
- Invalid formatting ("The group name may only contain ASCII letters and digits.")
  - I borrowed the "ASCII letters" thing from GitHub's repository creation form, though maybe that's equally as technical as regex stuff. Feel free to workshop that.

Reply:
Looks like this was resolved in 24fee91.